app/controllers/catalog_controller.rb

```diff
@@ -1258,10 +1265,21 @@ def get_form_vars
     @tenants_tree = build_tenants_tree # Build the tree with available tenants
     @available_catalogs = available_catalogs.sort # Get available catalogs with tenants and ancestors
     @additional_tenants = @edit[:new][:tenant_ids].map(&:to_s) # Get ids of selected Additional Tenants in the Tenants tree
+
+    if params[:currency]
+      @edit[:new][:currency] = params[:currency].to_i
+      @edit[:new][:code_currency] = code_currency_label(params[:currency])
+    end
+    @edit[:new][:price] = params[:price]
```

Comment:
missing `if params[:price]`?

This is called both when saving from angular (`st_edit` with `?button=save` or `add`), and from non-angular `st_form_field_changed`.